File: daily-challenge/2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.java
class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {
        HashSet<Integer> hs = new HashSet<>((int) 1e4);
        for (int i : banned)
            hs.add(i);
        int sum = 0;
        int picked = 0;
        for (int i = 1; i <= n; i++) {
            if (hs.contains(i))
                continue;
            if (sum + i <= maxSum) {
                sum += i;
                picked++;
            } else
                return picked;
        }
        return picked;
    }
}

File: daily-challenge/2640-maximum-number-of-integers-to-choose-from-a-range-i/maximum-number-of-integers-to-choose-from-a-range-i.cs
public class Solution {
    public int MaxCount(int[] banned, int n, int maxSum) {
HashSet<int> b = new HashSet<int>();

foreach(int a in banned)
{
    b.Add(a);
}
        int count=0, sum=0;
        for(int i=1; i<= n; i++)
        {
            if(!b.Contains(i))
            {
                
                sum += i;
                if(sum <= maxSum)
                {
                    count++;
                }
            }
        }
        return count;
        
    }
}